File: python/beeai_framework/adapters/a2a/agents/events.py
# Copyright 2025 © BeeAI a Series of LF Projects, LLC
# SPDX-License-Identifier: Apache-2.0

from typing import Any

from pydantic import BaseModel


class A2AAgentUpdateEvent(BaseModel):
    value: dict[str, Any]


class A2AAgentErrorEvent(BaseModel):
    message: str


a2a_agent_event_types: dict[str, type] = {
    "update": A2AAgentUpdateEvent,
    "error": A2AAgentErrorEvent,
}

File: typescript/src/adapters/a2a/agents/events.ts
/**
 * Copyright 2025 © BeeAI a Series of LF Projects, LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { Callback } from "@/emitter/types.js";
import { Message, Task, TaskArtifactUpdateEvent, TaskStatusUpdateEvent } from "@a2a-js/sdk";

export interface A2AAgentUpdateEvent {
  value: Message | Task | TaskStatusUpdateEvent | TaskArtifactUpdateEvent;
}

export interface A2AAgentErrorEvent {
  message: string;
}

export interface A2AAgentEvents {
  update?: Callback<A2AAgentUpdateEvent>;
  error?: Callback<A2AAgentErrorEvent>;
}
